### ISSUE TITLE
Icons don't always appear/disappear in taskbar when app opened/closed

### ISSUE THREAD

**Issue by ghost** [jderose9/dash-to-panel#96]
The icons do not show up in panel after launching for sometime. The icons are however persisted for some time even after that application has been closed.

**Comment by jderose9** [comment]
So you see this intermittently? Is this only with specific applications? Which version of gnome and distro are you using?

I've seen this when there are multiple .desktop files for the same binary with different command line parameters.. ie. LibreOffice. But, it is a bug in Gnome shell and can be seen in the Activities dash without this extension installed.

**Comment by RintarouOkabe** [comment]
Thanks for replying.

I see this problem with Archive Manager, Sublime Text, Gedit and Chromium. I an using Ubuntu GNOME 16.04 LTS.

The application are visible in Activities, but not on the panel.

**Comment by RintarouOkabe** [comment]
I tried checking all extensions and seems like 'Dynamic Top Bar' is responsible for the delay in launch and removal of icons. I will check more thoroughly and report further.

**Comment by jderose9** [comment]
Can you check out [Dynamic Panel Transparency](https://extensions.gnome.org/extension/1011/dynamic-panel-transparency/) ? It seems to do the same thing and I know that a lot of people are using it in combination with this extension.

**Comment by RintarouOkabe** [comment]
The issue returns when using Dynamic Panel Transparency. :|

**Comment by jderose9** [comment]
I tried 16.04 with Dynamic Panel Transparency and don't see the issue. Can you try with the most recent release of Dynamic Panel Transparency as I believe they just fixed some rendering issues?

Could you also check the logs? You can open a terminal and type this:
`journalctl /usr/bin/gnome-shell -f -o cat`

Then try opening one of the offending applications and see if any errors appear when the application fails to appear on the taskbar.

**Comment by kilbith** [comment]
Arch Linux. No Dynamic Panel Transparency.

**Comment by franglais125** [comment]
@jderose9 I don't mean to spam/divert attention, just a thought:

if something is interfering with the redisplay queue, we can change some `this._queueRedisplay` calls with `this._redisplay`, so that the call doesn't get blocked in the queue for whatever reason.

This is not "solving" the problem though, just working around it (assuming it even works).

**Comment by jderose9** [comment]
@franglais125 I was initially thinking that the onFocusAppChanged had a bug that only appeared under some specific configuration or that a specific version of gnome wasn't firing the event it was listening to consistently.

However, you may be right. In that case, it would sound similar to #92.

Whenever I've seen things simply fail to draw, either the actor its being drawn on hasn't been initialized yet, or I see this error hit the log:
`_cogl_buffer_fini: assertion '!(buffer->flags & COGL_BUFFER_FLAG_MAPPED)' failed`

In those cases something was being drawn immediately rather than queued (the reverse of what you are describing)... but that's just issues I dealt with in the past.

It would be great if we could figure out reproduction steps, but hopefully someone here can provide a log at least.

**Comment by serenase** [comment]
Here is my log.
`journalctl /usr/bin/gnome-shell -f -o cat
st_widget_get_theme_node called on the widget [0x52a9f60 StBin:last-child first-child] which is not in the stage.
st_widget_get_theme_node called on the widget [0x6e11830 Gjs_DashItemContainer.dash-item-container:last-child ("7")] which is not in the stage.
st_widget_get_theme_node called on the widget [0x52a5fe0 StButton.app-well-app running1:last-child first-child ("7")] which is not in the stage.
st_widget_get_theme_node called on the widget [0x52a74b0 StWidget:last-child first-child ("7")] which is not in the stage.
st_widget_get_theme_node called on the widget [0x52a7fc0 StBin.overview-icon] which is not in the stage.
st_widget_get_theme_node called on the widget [0x52a8a80 ShellGenericContainer:last-child first-child] which is not in the stage.
st_widget_get_theme_node called on the widget [0x52a9f60 StBin:last-child first-child] which is not in the stage.
st_widget_get_theme_node called on the widget [0x6e147d0 StIcon:last-child first-child] which is not in the stage.
/usr/lib/gio/modules/libgvfsdbus.so: file too short
Failed to load module: /usr/lib/gio/modules/libgvfsdbus.so
`

**Comment by serenase** [comment]
Here's mine...

`journalctl /usr/bin/gnome-shell -f -o cat
st_widget_get_theme_node called on the widget [0x52a9f60 StBin:last-child first-child] which is not in the stage.
st_widget_get_theme_node called on the widget [0x6e11830 Gjs_DashItemContainer.dash-item-container:last-child ("7")] which is not in the stage.
st_widget_get_theme_node called on the widget [0x52a5fe0 StButton.app-well-app running1:last-child first-child ("7")] which is not in the stage.
st_widget_get_theme_node called on the widget [0x52a74b0 StWidget:last-child first-child ("7")] which is not in the stage.
st_widget_get_theme_node called on the widget [0x52a7fc0 StBin.overview-icon] which is not in the stage.
st_widget_get_theme_node called on the widget [0x52a8a80 ShellGenericContainer:last-child first-child] which is not in the stage.
st_widget_get_theme_node called on the widget [0x52a9f60 StBin:last-child first-child] which is not in the stage.
st_widget_get_theme_node called on the widget [0x6e147d0 StIcon:last-child first-child] which is not in the stage.
/usr/lib/gio/modules/libgvfsdbus.so: file too short
Failed to load module: /usr/lib/gio/modules/libgvfsdbus.so
Could not load a pixbuf from icon theme.
This may indicate that pixbuf loaders or the mime database could not be found.
Archive:  /tmp/3VQFYY.shell-extension.zip
   creating: /home/satu/.local/share/gnome-shell/extensions/suspend-button@laserb/schemas/
   creating: /home/satu/.local/share/gnome-shell/extensions/suspend-button@laserb/locale/
   creating: /home/satu/.local/share/gnome-shell/extensions/suspend-button@laserb/locale/de_CH/
   creating: /home/satu/.local/share/gnome-shell/extensions/suspend-button@laserb/locale/pl/LC_MESSAGES/
  inflating: /home/satu/.local/share/gnome-shell/extensions/suspend-button@laserb/locale/zh_CN/LC_MESSAGES/gnome-shell-extension-suspend-button.po
  inflating: /home/satu/.local/share/gnome-shell/extensions/suspend-button@laserb/schemas/gschemas.compiled
  inflating: /home/satu/.local/share/gnome-shell/extensions/suspend-button@laserb/README.md
  inflating: /home/satu/.local/share/gnome-shell/extensions/suspend-button@laserb/locale/pl/LC_MESSAGES/gnome-shell-extension-suspend-button.mo
  inflating: /home/satu/.local/share/gnome-shell/extensions/suspend-button@laserb/locale/de_CH/LC_MESSAGES/gnome-shell-extension-suspend-button.po
  inflating: /home/satu/.local/share/gnome-shell/extensions/suspend-button@laserb/locale/zh_CN/LC_MESSAGES/gnome-shell-extension-suspend-button.mo
  inflating: /home/satu/.local/share/gnome-shell/extensions/suspend-button@laserb/schemas/org.gnome.shell.extensions.suspend-button.gschema.xml
  inflating: /home/satu/.local/share/gnome-shell/extensions/suspend-button@laserb/lib.js
   creating: /home/satu/.local/share/gnome-shell/extensions/suspend-button@laserb/locale/de/LC_MESSAGES/
  inflating: /home/satu/.local/share/gnome-shell/extensions/suspend-button@laserb/locale/pl/LC_MESSAGES/gnome-shell-extension-suspend-button.po
  inflating: /home/satu/.local/share/gnome-shell/extensions/suspend-button@laserb/COPYING
  inflating: /home/satu/.local/share/gnome-shell/extensions/suspend-button@laserb/locale/de/LC_MESSAGES/gnome-shell-extension-suspend-button.po
  inflating: /home/satu/.local/share/gnome-shell/extensions/suspend-button@laserb/locale/de/LC_MESSAGES/gnome-shell-extension-suspend-button.mo
  inflating: /home/satu/.local/share/gnome-shell/extensions/suspend-button@laserb/locale/de_CH/LC_MESSAGES/gnome-shell-extension-suspend-button.mo
  inflating: /home/satu/.local/share/gnome-shell/extensions/suspend-button@laserb/extension.js
  inflating: /home/satu/.local/share/gnome-shell/extensions/suspend-button@laserb/prefs.js
 extracting: /home/satu/.local/share/gnome-shell/extensions/suspend-button@laserb/metadata.json
clutter_actor_remove_child: assertion 'child->priv->parent != NULL' failed
`
So it conflicts with status menu button extension what become preinstalled with light gdm, am I right? I turned it off and change tit to this similar extension [https://extensions.gnome.org/extension/826/suspend-button/](url) and everything seems to be fine.

**Comment by jderose9** [comment]
I'm seeing this on Antergos with the status menu extension installed, thanks!

**Comment by grandixximo** [comment]
Having same issue on Ubuntu gnome 16.04, i don't have Dynamic Panel Transparency installed, i can say tough that the issue seems to start right after i open and close nautilus if it's pinned as a favorite, and it behaves normally after a restart.

**Comment by grandixximo** [comment]
how can i provide a log? not sure how to start the shell from command

**Comment by imhotepisinvisible** [comment]
I'm seeing the same issue.  All dash icons take 10-20 seconds to appear every time I unlock.  Whenever I open a new app, e.g. gedit, it takes 10-20 seconds for the icon to appear.  Running the paper theme and using some other extensions, although the behavior seems to continue when I disable all other extensions and switch to default theme.

**Comment by jderose9** [comment]
I saw this at one point on a VM with Antergos but I no longer have the device and can't get it to happen with a current installation of Antergos on my current hardware. If anyone has any additional info on how to reproduce or hardware specifics needed to observe this it would be much appreciated.

@grandixximo 
>how can i provide a log? not sure how to start the shell from command

Just open a terminal window and type:
`journalctl /usr/bin/gnome-shell -f -o cat`
Then, take whatever action is required to reproduce the issue.

**Comment by imhotepisinvisible** [comment]
I'm pretty sure it's theme related.  If I have the default theme active then I don't have a problem.  I switch to arc theme and it happens consistently - every time I return from the lock screen and every time I open a new application there's a 10-20 second delay.  Nothing useful shows up in the log.

**Comment by darekdeo** [comment]
My mistake for creating the duplicate issue, I've been searching for this thread but couldn't find it.

I did not mention it in #194 but I am also using Arc theme (installed it while using Unity).

**Comment by sw9** [comment]
I've encountered a similar problem, but not sure if it's the same issue. Whenever I open a new application, the icon appears a little too far to the right. However, once I mouse over the icon, it jumps to the proper position.

**Comment by RintarouOkabe** [comment]
Can confirm. Installation from git doesn't pose this issue,

**Comment by Richy91** [comment]
Hello! I wanted to share with my experience with this issue!

I have been using this extension since ubuntu 16.04 and now on ubuntu 18.04 development branch.
On 16.04 after suspend icons always were missing for about ~15 to 20 seconds, but on 18.04 this issue for me is gone.

About opening and closing apps.
I use Arc-Darker theme and arc icons. I have noticed that if you open larger applications like chromium or codeblocks then icons appear with delay, but with smaller apps like calendar, calculator or text editor they appear fine. Usually icons appear and dissapear with little delay like 5 seconds for chromium and codeblocks, but sometimes they stuck for longer time and then delay starts for all apps. 
I have tried this with default theme and still the same. 

This delay also happens with default ubuntu dock (on 18.04.   gnome-shell 3.26.2). 

If you restart system and dont open mentioned applications, the everything works fine. Maybe this is not the extensions fault.

Anyway i hope this helps!
